saga/fx/take.ts

```diff
@@ -0,0 +1,23 @@
+import { action } from "../../instructions.ts";
+
+import type { Action } from "../types.ts";
+import { ActionContext } from "../context.ts";
+import { matcher } from "../matcher.ts";
+
+export function take(pattern: string[]) {
+  return action<Action>(function* (resolve) {
+    const actions = yield* ActionContext;
+    const msgs = yield* actions.output;
+    let next = yield* msgs;
+
+    while (!next.done) {
+      const match = matcher(pattern);
+      if (match(next.value)) {
+        resolve(next.value);
+        return;
+      }
+
+      next = yield* msgs;
+    }
+  });
+}
```

Comment:
The only time you really need to use `action()` is when you need to get a callback that you can hand out to an external system, like a `setTimeout` or an event listener. I think you can use a simple operation here.

```suggestion
export function* take(pattern: string[]) {
  const actions = yield* ActionContext;
  const msgs = yield* actions.output;
  let next = yield* msgs;

  while (!next.done) {
    const match = matcher(pattern);
    if (match(next.value)) {
      return next.value;
    }
    next = yield* msgs;
  }
}
```